timeseries/src/autogluon/timeseries/models/chronos/model.py

```diff
@@ -0,0 +1,366 @@
+        If True, validation will be skipped during training
+    device : str, default = None
+        Device to use for inference. If None, model will use the GPU if available. For larger model sizes
+        `small`, `base`, and `large`; inference will fail if no GPU is available.
+    optimization_strategy : str, default = None
+        Optimization strategy to use for inference on CPUs. If None, the model will use the default implementation.
+        If `onnx`, the model will be converted to ONNX and the inference will be performed using ONNX. If `ovm`,
+        inference will be performed with the model compiled to OpenVINO.
+    """
+
+    # default number of samples for prediction
+    default_num_samples: int = 20
+    default_batch_size: int = 16
+    default_context_length: int = 512
+    bucket_name = "chronos-release"
+    default_model_path = "amazon/chronos-t5-small"
+
+    def __init__(
+        self,
+        freq: Optional[str] = None,
+        prediction_length: int = 1,
+        path: Optional[str] = None,
+        name: Optional[str] = None,
+        eval_metric: str = None,
+        hyperparameters: Dict[str, Any] = None,
+        **kwargs,  # noqa
+    ):
+        hyperparameters = hyperparameters if hyperparameters is not None else {}
+        self.batch_size = hyperparameters.get("batch_size", self.default_batch_size)
+        self.num_samples = hyperparameters.get("num_samples", self.default_num_samples)
+        self.model_path = hyperparameters.get("model_path", self.default_model_path)
+        self.skip_validation = hyperparameters.get("skip_validation", False)
+        self.device = hyperparameters.get("device", None)
+        self.optimization_strategy: Optional[Literal["onnx", "ovm"]] = hyperparameters.get(
+            "optimization_strategy", None
+        )
+        self.context_length = hyperparameters.get("context_length", self.default_context_length)
+
+        model_path_safe = str.replace(self.model_path, "/", "__")
+        name = (name if name is not None else "Chronos") + f"[{model_path_safe}]"
+
+        super().__init__(
+            path=path,
+            freq=freq,
+            prediction_length=prediction_length,
+            name=name,
+            eval_metric=eval_metric,
+            hyperparameters=hyperparameters,
+            **kwargs,
+        )
+
+        self.model_pipeline: Optional[OptimizedChronosPipeline] = None
+
+    def save(self, path: str = None, verbose: bool = True) -> str:
+        pipeline = self.model_pipeline
+        self.model_pipeline = None
+        path = Path(super().save(path=path, verbose=verbose))
+        self.model_pipeline = pipeline
+
+        return str(path)
+
+    @classmethod
+    def load(cls, path: str, reset_paths: bool = True, verbose: bool = True) -> "ChronosModel":
+        model = load_pkl.load(path=os.path.join(path, cls.model_file_name), verbose=verbose)
+        model.model_pipeline = model._get_model_pipeline()
```

Comment:
Just to check my understanding, will this load the weights to GPU memory immediately? Or are they only loaded at prediction time?

Reply:
It loads them immediately.